test/unit/puppet_fact_parser_test.rb

```diff
@@ -233,8 +234,35 @@ def setup
     assert_equal '172.30.43.0', parser.interfaces['local_area_connection_2'][:network]
   end
 
+  test "should create connection between new os and new arch on facts parsing" do
+    assert os_to_arch_connectivity_check
+  end
+
+  test "should create connection between existing os and new arch on facts parsing" do
+    FactoryGirl.create(:operatingsystem,:name => 'AnyOS', :major => '6')
+    assert os_to_arch_connectivity_check
```

Comment:
I'm a little uneasy with how these tests create an object but don't actually check that the importer used the OS/arch that you created.

I'd suggest refactoring a little and using the result from FactoryGirl.create in your assertion, so we know it used your pre-created one, rather than only using importer.os and importer.arch.  For example:

<pre>
describe "OS/arch association" do
  setup do
    @importer = PuppetFact....
  end

  test "should create connection between existing os and new arch on facts parsing" do
    os = FactoryGirl.create(:operatingsystem,:name => 'AnyOS', :major => '6')
    assert_includes importer.architecture.operatingsystems(os)
  end
end
</pre>

This should also remove the need for an extra method, and puts it all in one describe block.